Supports ESP32-S3 platform

vikramdattu opened the pull request:
This PR targets:
 - Support Latest IDF versions
 - Code cleanup
 - Some critical fixes
 - video capture/encode for streaming

Top level README.md is updated with the esp32s3 support

Related issues: #36

vikramdattu committed:
Added ESP32-S3-EYE support

 - Use feed from camera instead of SD card
 - Encode frames using esp_h264 encoder
 - added main/idf_component.yml to resolve dependencies
 - Removed sdkconfig, added sdkconfig.defaults and sdkconfig.defaults.esp32s3

Image resolution: 320x240
FPS achieved is ~10

vikramdattu committed:
Fixes to get KVS demo working on IDF master

 - Fixed wrong usage of SNPRINTF
 - Fixed linking issues when ENABLE_STREAMING is switched off
 - Format specifier fixes in printf
 - Fixes on h264 frame size: Encoder can emit larger frames
 - Improvements in memory tracking: print mem stats at important locations

vikramdattu committed:
Multiple fixes and improvements

 - spiffs support
 - spiffs_image/ folder will contain folders such as certs for iot credentials and samples/
 - Some patches to fix the example on latest IDFs

vikramdattu committed:
Use netif_sntp to avoid race conditions.

Also, increased ntp servers to 2

vikramdattu committed:
Added CLI commands for debugging

 - Contains multiple console commands to aid debugging. Can be queried with `help` command
 - Also added wifi-set CLI: Can be used to set wifi configs on the fly. Device reboots into new config.

vikramdattu committed:
Added patch for multiple issue fixes

 - Safe link list traversal
 - Fix setup iot credentials script
 - MBEDTLS 3.x support
 - Format specifiers for printf fixes

vikramdattu committed:
Add video_enc_task which will work asynchronous to encode frames

vikramdattu committed:
Enabled audio send support

 - Sends pre-encoded audio frames
 - I2S recording and OPUS encoding not supported yet

vikramdattu committed:
Improved README.md

 - Include ESP32-S3 in reference platforms list
 - Some minor fixes

vikramdattu committed:
Change to switch between older vs new h264

vikramdattu committed:
Added OPUS encoder which encodes i2s audio stream

 - Special task for OPUS encoder (similar to H264) is created which keeps encoding audio frames

vikramdattu commented:
cc @ycyang1229 @codingspirit

vikramdattu committed:
Fix DEFAULT_KVS_CACERT_PATH path

 - Needs a change in `include/kvs/common.h`
 - `#define DEFAULT_KVS_CACERT_PATH "/spiffs/certs/cacert.pem"`

codingspirit commented:
Hey @vikramdattu, I'm no longer maintaining this repository. @awsblake Please review this PR

ycyang1229 commented:
I no longer work for AWS, so it is better to reach out other AWS guys.

ycyang1229 commented:
I no longer work for AWS. It it better to reach out to other AWS members since this is one AWS repository. But if you want to develop one official branch in Espressif repository, I will be interested. Thanks.

vikramdattu committed:
Fix wrong header includes `freeRTOS.h` --> `FreeRTOS.h`

 - Worked on non-case sensitive platforms(Mac/Windows) hence gone unnoticed
 - Fixed this by using exact case sensitive file names
